Sdk/MoyasarSdk/Views/STCPayModule/View/STCPayView.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  STCPayView.swift
+//  MoyasarSdk
+//
+//  Created by Mahmoud Abdelwahab on 11/09/2024.
+//
+
+import SwiftUI
+
+public struct STCPayView: View {
+    /// The observed view model that handles the STC Pay logic.
+    @ObservedObject var viewModel: STCPayViewModel
+    
+    /// Initializes the `STCPayView` with a payment request and a callback for handling the payment result.
+      ///
+      /// - Parameters:
+      ///   - paymentRequest: The request containing payment details, including the API key.
+      ///   - callback: A closure that gets called when the payment process finishes with success or failure.
+      ///
+      /// - Throws: If the `STCPayViewModel` initialization fails due to an invalid API key,
+      /// it throws an error, which is handled by showing a fatal error message.
+    public init(paymentRequest: PaymentRequest, callback: @escaping STCResultCallback) {
+        do {
+            viewModel = try STCPayViewModel(paymentRequest: paymentRequest, resultCallback: callback)
+        } catch {
+            // Handle error here, show error in view model
+            fatalError("Invalid api key 🙁")
+        }
+    }
+    
+    public var body: some View {
+        VStack {
+            switch viewModel.screenStep {
+            case .mobileNumber:
+                mobileNumberView
+            case .otp:
+                otpView
+            }
+        }.padding()
+    }
+}
+
+struct STCPayView_Previews: PreviewProvider {
```

Comment:
I'm not familiar with Swift, but these look like hardcoded values to be removed, right?

Reply:
It is hard coded  must needed for Swift UI preview only

Reply:
So is it for the demo part of the SDK only? If yest, I think it's better to be under the demo namespace.

Reply:
It is for swift ui preview that show ui during the development recommended by apple  not related to the demo